python/pyarrow/tests/test_serialization.py

```diff
@@ -520,6 +537,107 @@ def deserializer(data):
     assert np.alltrue(new_x.view(np.ndarray) == np.zeros(3))
 
 
+@pytest.mark.skip
+def test_sparse_tensor_coo_components_serialization(large_buffer):
+    data = np.array([[1, 1, 1, 1, 1, 1, 1]]).T
+    row = np.array([0, 0, 1, 3, 1, 0, 0])
+    col = np.array([0, 2, 1, 3, 1, 0, 0])
+    shape = (4, 4)
+
+    coords = np.vstack([row, col]).T
+    sparse_tensor = pa.SparseTensorCOO.from_numpy(data, coords, shape)
+    serialization_roundtrip(sparse_tensor, large_buffer)
+
+
+def test_sparse_tensor_coo_serialization(large_buffer):
+    data = np.array([[1, 1, 1, 1, 1, 1, 1]]).T
+    row = np.array([0, 0, 1, 3, 1, 0, 0])
+    col = np.array([0, 2, 1, 3, 1, 0, 0])
+    shape = (4, 4)
+
+    coords = np.vstack([row, col]).T
+    sparse_tensor = pa.SparseTensorCOO.from_numpy(data, coords, shape)
+
+    serialized = pa.serialize(sparse_tensor)
+    result = serialized.deserialize()
+
+    data_result, coords_result = result.to_numpy()
+    assert np.array_equal(data_result, data)
+    assert np.array_equal(coords_result, coords)
+
+
+def test_sparse_tensor_csr_serialization(large_buffer):
+    data = np.array([[1, 2, 3, 4, 5, 6]]).T
+    indptr = np.array([0, 2, 3, 6])
+    indices = np.array([0, 2, 2, 0, 1, 2])
+    shape = (3, 3)
+
+    sparse_tensor = pa.SparseTensorCSR.from_numpy(data, indptr, indices, shape)
+
+    serialized = pa.serialize(sparse_tensor)
+    result = serialized.deserialize()
+
+    data_result, indptr_result, indices_result = result.to_numpy()
+    assert np.array_equal(data_result, data)
+    assert np.array_equal(indptr_result, indptr)
+    assert np.array_equal(indices_result, indices)
+
+
+@pytest.mark.skipif(os.name == 'nt', reason="equality check fails on windows")
```

Comment:
Do you know why the equality check fails on Windows?

Reply:
I can't tell from the error log. Here's the same failure from ARROW-4224 [error log](https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/27800378/job/7gngqe4o76qsw8js#L1981). It could be an issue within `arrow/compare.cc`?

Unfortunately I don't have a Windows machine to test on it.
I might eventually get a VM to check this on but that won't happen soon.

Reply:
Well, it should be solved before even thinking about integrating this PR, I'd say. I may give a hand at some point but it's low-priority for me.

Reply:
Agreed.
If it is a problem within `arrow/compare.cc` then it's already in master.

Reply:
I have a Windows VM so I can investigate it.
But I haven't created the development environment on Windows, so I probably need several days to create the environment for the investigation.

Reply:
It appears `test_sparse_tensor_coo_components_serialization` also [fails on windows](https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/27818470/job/e1qfag981u02mmje#L1975).

Reply:
I've set up a VM and will look into the issue.

Reply:
It's a deserialization issue. I'm working on it.
@mrkn referring to https://github.com/apache/arrow/pull/4990#discussion_r332347384 - I have some time to work on it this week and can take it off your hands. Or do you prefer to do it?

Reply:
@rok Sorry I kept you waiting.  I've finished some refactoring works.

Reply:
Thanks @mrkn!

Reply:
I've rebased to this branch and resolved the index deserialization issue. I'll work on the data (values) deserialization tomorrow. Thanks again for the help @mrkn! :)